graphql_compiler/compiler/emit_sql.py

```diff
@@ -93,11 +110,14 @@ def __init__(self, sql_schema_info, ir):
         # Metadata
         self._sql_schema_info = sql_schema_info
         self._ir = ir
-        self._used_columns = _find_columns_used(sql_schema_info, ir)
+        self._used_columns = _find_columns_used_outside_folds(sql_schema_info, ir)
+        self._all_folded_outputs = _find_folded_outputs(ir)
+        self._in_fold = False
```

Comment:
This is not immutable metadata, but very mutable state that depends on the current progress along the ir. I'd move it lower down, and maybe rename `# Metadata` to `# Immutable metadata` to be more explicit.